hyperspy/io_plugins/sur.py

```diff
@@ -0,0 +1,1298 @@
+            data.append(obj['_62_points'].reshape(shape))
+
+        #Pushing data into the dictionary
+        self.signal_dict['data'] = np.stack(data)
+
+    def _build_RGB_image(self,):
+        """Build an RGB image. The T axis is navigation and set from
+        P Size"""
+
+        #First object dictionary
+        hypdic = self._list_sur_file_content[0]
+
+        #Metadata are set from first dictionary
+        self.signal_dict['metadata'] = self._build_metadata(hypdic)
+        self.signal_dict['original_metadata'] = self._build_original_metadata()
+
+        #Add the series-axis to the signal dict
+        self.signal_dict['axes'].append(\
+            self._build_Tax(hypdic,'_08_P_Size',ind=0,nav=True))
+
+        #All objects must share the same signal axes
+        self.signal_dict['axes'].append(\
+                self._build_Yax(hypdic,ind=1,nav=False))
+        self.signal_dict['axes'].append(\
+            self._build_Xax(hypdic,ind=2,nav=False))
+
+        #shape of the surfaces in the series
+        shape = (hypdic['_19_Number_of_Lines'],hypdic['_18_Number_of_Points'])
+        #We put all the data together
+        data = []
+        for obj in self._list_sur_file_content:
+            data.append(obj['_62_points'].reshape(shape))
+
+        #Pushing data into the dictionary
+        self.signal_dict['data'] = np.stack(data)
+
+        self.signal_dict.update({'post_process':[self.post_process_RGB]})
+
+    ### Metadata utility methods
+    def _build_metadata(self,unpacked_dict):
+        """Return a minimalistic metadata dictionary according to hyperspy
+        format. Accept a dictionary as an input because dictionary with the
+        headers of a mountians object.
+
+        Parameters
+        ----------
+        unpacked_dict: dictionary from the header of a surface file
+
+        Returns
+        -------
+        metadict: dictionnary in the hyperspy metadata format
+
+        """
+
+        #Formatting for complicated strings
+        quantity_str = " ".join([
+            unpacked_dict['_26_Name_of_Z_Axis'],
+            unpacked_dict['_29_Z_Step_Unit'],
+            ]).strip()
+
+        date_str =  "{:4d}-{:2d}-{:2d}".format( \
+            unpacked_dict['_45_Year'],
+            unpacked_dict['_44_Month'],
+            unpacked_dict['_43_Day'])
+
+        time_str = "{:2d}:{:2d}:{:2d}".format( \
+            unpacked_dict['_42_Hours'],
+            unpacked_dict['_41_Minutes'],
+            unpacked_dict['_40_Seconds'])
+
+        #Metadata dictionary initialization
+        metadict = {
+            "General":{
+                "authors": unpacked_dict['_07_Operator_Name'],
+                "date":date_str,
+                "original_filename": os.path.split(self.filename)[1],
+                "time": time_str,
+                },
+            "Signal": {
+                "binned": False,
+                "quantity": quantity_str,
+                "signal_type": "",
```

Comment:
This is a bit difficult for the moment. The problem is that this data format does not make a difference between different spectra, e.g. Raman, CL, PL. This is also true for hyperspectral maps and surfaces (can be a CL image integrated on energy, loadings of a decomposed map etc... ).

Although this could be linked, in some cases, with lumispy types, I must admit that I would rather let users set the signal type themselves

Reply:
Sounds very sensible!